src/github/client/app-token-holder.ts

```diff
@@ -5,14 +5,35 @@ import { AuthToken, ONE_MINUTE, TEN_MINUTES } from "./auth-token";
 import * as PrivateKey from "probot/lib/private-key";
 import LRUCache from "lru-cache";
 import { InstallationId } from "./installation-id";
+import { Subscription } from "~/src/models/subscription";
+import { GitHubServerApp } from "~/src/models/github-server-app";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
 
-export type KeyLocator = (installationId: InstallationId) => string;
+export type KeyLocator = (installationId: InstallationId) => Promise<string>;
 
 /**
  * By default, we just look for a key in the `PRIVATE_KEY` env var.
  */
 export const cloudKeyLocator: KeyLocator = () => {
-	return PrivateKey.findPrivateKey() || "";
+	const privateKey = PrivateKey.findPrivateKey() || "";
+	return Promise.resolve(privateKey);
+};
+
+/**
+ * Look for a private key in GitHubServerApp table
+ */
+export const gheKeyLocator = async (gitHubInstallation: InstallationId) => {
```

Comment:
It is bit complicated. The `GHE` key locator return a `Cloud` key locator. 
I guess you are doing this because we are using `singleton` pattern on the `AppTokenHolder`?

if so, I think maybe rename the `gheKeyLocator` to `keyLocator` or `defaultKeyLocator`

Reply:
I don't think we need 2 different key locators functions, it should just be the one function based on the parameters given.

Reply:
I'd also probably move the key locator stuff into it's own util file and it's own test

Reply:
@mboudreau , moved key locator into seperate file and test as well.